internal/pkg/cli/dialog/use_template.go

```diff
@@ -0,0 +1,107 @@
+package dialog
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/prompt"
+	"github.com/keboola/keboola-as-code/internal/pkg/template/input"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils"
+)
+
+type contextKey string
+
+func (p *Dialogs) AskUseTemplateOptions(inputs input.Inputs) (results map[string]interface{}, err error) {
+	results = make(map[string]interface{})
+	ctx := context.Background()
+	errors := utils.NewMultiError()
+	for _, i := range inputs {
+		if !i.Available(results) {
+			continue
+		}
+		switch i.Kind {
+		case input.KindInput, input.KindPassword, input.KindTextarea:
+			question := &prompt.Question{
+				Label:       i.Name,
+				Description: i.Description,
+				Validator: func(raw interface{}) error {
+					value, err := convertType(raw, i.Type)
+					if err != nil {
+						return err
+					}
+					return i.ValidateUserInput(value, ctx)
+				},
+				Hidden: i.Kind == input.KindPassword,
+			}
+			if i.Default != nil {
+				question.Default = i.Default.(string)
+			}
+			value, _ := p.Ask(question)
+			ctx = context.WithValue(ctx, contextKey(i.Id), value)
```

Comment:
Moc se mi tady nezdá, jak se výsledky ukládají současně do kontextu i do `results`, ale nenašel jsem způsob jak z toho `context` objektu dostat mapu těch hodnot, tak snad je to takhle ok.

Reply:
jj, z `context` sa neda citat, ... IMHO je to ok, neda sa inac, potrebujes oba reprezentacie, pre 2 kniznice.